src/transformStateWithClones.js

```diff
@@ -7,7 +7,24 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    if (action.type === `addProperties`) {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === `removeProperties`) {
+      action.keysToRemove.forEach((key) => {
+        delete currentState[key];
+      });
+    } else if (action.type === `clear`) {
+      currentState = {};
+    }
```

Comment:
[NAMING] - The name `currentState` for the object copy is appropriate, but ensure that you maintain this naming convention consistently throughout your code to improve readability.